common/cards/card-plugins/hermits/evilxisuma_rare.js

```diff
@@ -0,0 +1,136 @@
+				power: null,
+			},
+			secondary: {
+				name: 'Derpcoin',
+				cost: ['balanced', 'balanced'],
+				damage: 80,
+				power:
+					"Flip a coin.\n\nIf heads, choose one of the opposing active Hermit's attacks to disable on their next turn.",
+			},
+		})
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {CardPos} pos
+	 */
+	onAttach(game, instance, pos) {
+		const {player, otherPlayer} = pos
+
+		player.hooks.onAttack[instance] = (attack, pickedSlots) => {
+			if (
+				attack.id !== this.getInstanceKey(instance) ||
+				attack.type !== 'secondary'
+			)
+				return
+
+			const coinFlip = flipCoin(player, this.id)
+			player.coinFlips[this.id] = coinFlip
+
+			if (coinFlip[0] !== 'heads') return
+
+			player.followUp = this.id
+
+			// He only disables the attack of the target, that means that
+			// lightning rod counters him and using knockback/target block
+			// is a really bad idea
+			player.custom[this.getInstanceKey(instance, 'target')] =
+				attack.target.index
+
+			// It's easier to not duplicate the code if I use the hooks here
+			player.hooks.onFollowUp[instance] = (
+				followUp,
+				pickedSlots,
+				modalResult
+			) => {
+				if (followUp !== this.id) return
+				delete player.hooks.onFollowUp[instance]
+				delete player.hooks.onFollowUpTimeout[instance]
+				player.followUp = null
+
+				if (!modalResult || !modalResult.disable) return
+
+				player.custom[this.getInstanceKey(instance, 'disable')] =
+					modalResult.disable
+			}
+
+			player.hooks.onFollowUpTimeout[instance] = (followUp) => {
+				if (followUp !== this.id) return
+				delete player.hooks.onFollowUpTimeout[instance]
+				delete player.hooks.onFollowUpTimeout[instance]
+				player.followUp = null
+
+				// Disable the secondary attack if the player didn't choose one
+				player.custom[this.getInstanceKey(instance, 'disable')] = 'secondary'
+			}
+
+			otherPlayer.hooks.blockedActions[instance] = (blockedActions) => {
+				const disable = player.custom[this.getInstanceKey(instance, 'disable')]
+				const targetRow = player.custom[this.getInstanceKey(instance, 'target')]
+				if (!disable) return blockedActions
+
+				const activeRow = otherPlayer.board.activeRow
+				if (!activeRow || !targetRow) return blockedActions
+				if (activeRow !== targetRow) return blockedActions
+
+				blockedActions.push(
+					disable === 'primary' ? 'PRIMARY_ATTACK' : 'SECONDARY_ATTACK'
+				)
+
+				return blockedActions
+			}
+
+			otherPlayer.hooks.onTurnEnd[instance] = () => {
+				delete otherPlayer.hooks.blockedActions[instance]
+				delete otherPlayer.hooks.onTurnEnd[instance]
+				delete player.custom[this.getInstanceKey(instance, 'disable')]
+				delete player.custom[this.getInstanceKey(instance, 'target')]
+			}
+		}
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {CardPos} pos
+	 */
+	onDetach(game, instance, pos) {
+		const {player} = pos
+		delete player.hooks.onAttack[instance]
```

Comment:
why are you not deleting all the hooks?

Reply:
Not needed, onFollowUp and onFollowUpTimeout are created inside onAttack and they delete themselves as soon as they are called, onTurnEnd and blockedActions get deleted on the end of the opponents turn, I can’t delete them when Evil X dies from a backslash because the effect would stop early.